server/src/users/users.service.ts

```diff
@@ -26,34 +29,44 @@ export class UsersService {
     }
   }
 
-  findOne(id: number) {
-    return this.usersRepository.findOne(id);
+  async findOne(id: number): Promise<Omit<User, 'password'>> {
+    const user = await this.usersRepository.findOne(id);
+    delete user.password;
+    return user;
   }
 
-  // Search database for user with matching email.
-  // Returns user on success, throws 404 error if user does not exist
-  async findByEmail(email: string) {
+  async findByEmail(email: string, includePw = false): Promise<User | Omit<User, 'password'>> {
     const user = await this.usersRepository.findOne({ email });
+
     if (!user) {
       throw new HttpException(
         { status: HttpStatus.NOT_FOUND, error: 'Email not found' },
         HttpStatus.NOT_FOUND,
       );
     }
+
+    if (!includePw) {
+      delete user.password;
+    }
     return user;
   }
-  //Change to whatever the display name ends up being.
-  findByUsername(first_name: string) {
-    return this.usersRepository.findOne({ first_name });
+
+  // Change to whatever the display name ends up being.
+  async findByUsername(first_name: string): Promise<Omit<User, 'password'>> {
+    const user = await this.usersRepository.findOne({ first_name });
+    delete user.password;
+    return user;
   }
 
-  //TODO: Assess if there is a better way than making two requests.
+  // TODO: Assess if there is a better way than making two requests.
   async update(id: number, updateUserDto: UpdateUserDto) {
     await this.usersRepository.update(id, updateUserDto);
-    return this.usersRepository.findOne(id);
+    const user = await this.usersRepository.findOne(id);
+    delete user.password;
+    return user;
```

Comment:
now deleting pw at earliest point in all cases
(aka in here except for 1 case where we need to check pw in auth service. in that case we delete right after pw check)